### DISCUSSION TITLE
Small UI testing/ GH Workflow Quirks!

### DISCUSSION THREAD

**Discussion by jennxu23** [CS342/.github#33] [Technical Questions] [answered]
### In what area do you have a technical challenge?

Xcode, Simulator & Previews

### Description

Over the past week, we've encountered some hard-to-track-down bugs in our Onboarding and HealthKitUpload UI tests.

### Reproduction

The two main issues we were running into were

1) Data wasn't displaying in the Mock Upload tab, which caused our HealthKitUpload UI test to fail. However, our data appeared to be connected to an account in Firebase, as we were auto logged-in in the XCode simulator.

2) Once we resolved the above issue, our tests passed locally. However, the Onboarding Flow would fail in the GitHub workflow with the following error message:

![IMG_5669](https://user-images.githubusercontent.com/56773304/221064004-775363ee-da6c-4aa9-9a62-912ee836311e.png)

### Expected behavior

1. When we started a new instance of a firebase emulator, the data from the previous instance would obviously be lost. However, on the XCode Simulator, we would auto-logged in to an account from the previous firebase instance. Therefore, none of our data was getting sent anywhere. To solve this issue, we cleared the contents of the XCode simulator before every build. .

2. In our onboarding flow, we moved around the order of some of the sheets (i.e we moved the consent view to after our sign up page). However, we didn't update the ordering in our AccountSetup module. The GitHub Simulator that's used in the GitHub workflow can't show the canvas view, and there's some code in the original application to work around that. Because of the disconnect between our onboarding flow and GitHub's expected onboarding flow, our tests would pass locally but fail on GitHub. To resolve this issue, we updated the following piece of code to reflect the correct order:

```
               #if targetEnvironment(simulator) && (arch(i386) || arch(x86_64))
                print("PKCanvas view-related views are currently skipped on Intel-based iOS simulators due to a metal bug on the simulator.")
                onboardingSteps.append(.accountSetup)
                #else
                onboardingSteps.append(.consent)
                #endif
```

### Code of Conduct

- [X] I agree to follow this course's Code of Conduct and Contributing Guidelines

**Comment by PSchmiedmayer** [comment, marked as answer]
Thank you @jennxu23 for sharing this with the class! 🚀
